Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -120,7 +146,84 @@ param (
     [System.Management.Automation.SwitchParameter]$SourceIsOffline
 )
 
-. $PSScriptRoot\..\Shared\OutputOverrides\Write-Host.ps1
+function Write-Host {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidOverwritingBuiltInCmdlets', '', Justification = 'Proper handling of write host with colors')]
+    [CmdletBinding()]
+    param(
+        [Parameter(Position = 1, ValueFromPipeline)]
+        [object]$Object,
+        [switch]$NoNewLine,
+        [string]$ForegroundColor
+    )
+    process {
+        $consoleHost = $host.Name -eq "ConsoleHost"
+
+        if ($null -ne $Script:WriteHostManipulateObjectAction) {
+            $Object = & $Script:WriteHostManipulateObjectAction $Object
+        }
+
+        $params = @{
+            Object    = $Object
+            NoNewLine = $NoNewLine
+        }
+
+        if ([string]::IsNullOrEmpty($ForegroundColor)) {
+            if ($null -ne $host.UI.RawUI.ForegroundColor -and
+                $consoleHost) {
+                $params.Add("ForegroundColor", $host.UI.RawUI.ForegroundColor)
+            }
+        } elseif ($ForegroundColor -eq "Yellow" -and
+            $consoleHost -and
+            $null -ne $host.PrivateData.WarningForegroundColor) {
+            $params.Add("ForegroundColor", $host.PrivateData.WarningForegroundColor)
+        } elseif ($ForegroundColor -eq "Red" -and
+            $consoleHost -and
+            $null -ne $host.PrivateData.ErrorForegroundColor) {
+            $params.Add("ForegroundColor", $host.PrivateData.ErrorForegroundColor)
+        } else {
+            $params.Add("ForegroundColor", $ForegroundColor)
+        }
+
+        Microsoft.PowerShell.Utility\Write-Host @params
+
+        if ($null -ne $Script:WriteHostDebugAction -and
+            $null -ne $Object) {
+            &$Script:WriteHostDebugAction $Object
+        }
+    }
+}
+
+function SetProperForegroundColor {
+    $Script:OriginalConsoleForegroundColor = $host.UI.RawUI.ForegroundColor
+
+    if ($Host.UI.RawUI.ForegroundColor -eq $Host.PrivateData.WarningForegroundColor) {
+        Write-Verbose "Foreground Color matches warning's color"
+
+        if ($Host.UI.RawUI.ForegroundColor -ne "Gray") {
+            $Host.UI.RawUI.ForegroundColor = "Gray"
+        }
+    }
+
+    if ($Host.UI.RawUI.ForegroundColor -eq $Host.PrivateData.ErrorForegroundColor) {
+        Write-Verbose "Foreground Color matches error's color"
+
+        if ($Host.UI.RawUI.ForegroundColor -ne "Gray") {
+            $Host.UI.RawUI.ForegroundColor = "Gray"
+        }
+    }
+}
+
+function RevertProperForegroundColor {
+    $Host.UI.RawUI.ForegroundColor = $Script:OriginalConsoleForegroundColor
+}
+
+function SetWriteHostAction ($DebugAction) {
+    $Script:WriteHostDebugAction = $DebugAction
+}
+
+function SetWriteHostManipulateObjectAction ($ManipulateObject) {
+    $Script:WriteHostManipulateObjectAction = $ManipulateObject
+}
```

Comment:
Revert back to the previous line. We want to have it dot loaded in the source code, the build pipeline will handle placing this into the file.

Reply:
done